lib/internal/http2/util.js

```diff
@@ -350,6 +354,67 @@ function getSettings(session, remote) {
 
 function updateSettingsBuffer(settings) {
   let flags = 0;
+  let numCustomSettings = 0;
+
+  if (typeof settings.customSettings === 'object') {
+    const customSettings = settings.customSettings;
+    for (const setting in customSettings) {
+      const val = customSettings[setting];
+      if (typeof val === 'number') {
+        let set = false;
+        const nsetting = Number(setting);
+        if (nsetting < IDX_SETTINGS_FLAGS) {
+          set = true;
+          switch (nsetting) {
+            case IDX_SETTINGS_HEADER_TABLE_SIZE:
+              flags |= (1 << IDX_SETTINGS_HEADER_TABLE_SIZE);
+              settingsBuffer[IDX_SETTINGS_HEADER_TABLE_SIZE] =
+                val;
+              break;
+            case IDX_SETTINGS_ENABLE_PUSH:
+              flags |= (1 << IDX_SETTINGS_ENABLE_PUSH);
+              settingsBuffer[IDX_SETTINGS_ENABLE_PUSH] = val;
+              break;
+            case IDX_SETTINGS_INITIAL_WINDOW_SIZE:
+              flags |= (1 << IDX_SETTINGS_INITIAL_WINDOW_SIZE);
+              settingsBuffer[IDX_SETTINGS_INITIAL_WINDOW_SIZE] =
+                val;
+              break;
+            case IDX_SETTINGS_MAX_FRAME_SIZE:
+              flags |= (1 << IDX_SETTINGS_MAX_FRAME_SIZE);
+              settingsBuffer[IDX_SETTINGS_MAX_FRAME_SIZE] =
+                val;
+              break;
+            case IDX_SETTINGS_MAX_CONCURRENT_STREAMS:
+              flags |= (1 << IDX_SETTINGS_MAX_CONCURRENT_STREAMS);
+              settingsBuffer[IDX_SETTINGS_MAX_CONCURRENT_STREAMS] = val;
+              break;
+            case IDX_SETTINGS_MAX_HEADER_LIST_SIZE:
+              flags |= (1 << IDX_SETTINGS_MAX_HEADER_LIST_SIZE);
+              settingsBuffer[IDX_SETTINGS_MAX_HEADER_LIST_SIZE] =
+                val;
+              break;
+            case IDX_SETTINGS_ENABLE_CONNECT_PROTOCOL:
+              flags |= (1 << IDX_SETTINGS_ENABLE_CONNECT_PROTOCOL);
+              settingsBuffer[IDX_SETTINGS_ENABLE_CONNECT_PROTOCOL] = val;
+              break;
+            default:
+              set = false;
+              break;
+          }
+        }
+        if (!set) { // not supported
+          if (numCustomSettings === MAX_ADDITIONAL_SETTINGS)
+            throw new ERR_HTTP2_TOO_MANY_CUSTOMSETTINGS();
+          settingsBuffer[IDX_SETTINGS_FLAGS + 1 + 2 * numCustomSettings + 1] = nsetting;
```

Comment:
This should likely verify that `nsetting` is within the appropriate numeric range.

Reply:
So something like this?
```
if ( typeof value !== 'number' || 
          IDX_SETTINGS_ENABLE_CONNECT_PROTOCOL >= nsettings 
           || nsettings > 0xffff )
           throw new ERR_HTTP2_INVALID_SETTING_VALUE.RangeError(
            'HTTP Setting index', nsetting, IDX_SETTINGS_ENABLE_CONNECT_PROTOCOL, 0xffff);
```
Also the value, but I can only check if it is within 32bit integer range.

Reply:
No, I change my mind, I will use:
```
        if (Number.isNaN(setting) ||
          typeof nsettings !== 'number' ||
          0 >= nsettings
          || nsettings > 0xffff)
          throw new ERR_HTTP2_INVALID_SETTING_VALUE.RangeError(
            'HTTP Setting index', nsetting, 0, 0xffff);
```
and combine it with the test for nan above.